### PR TITLE
refactor: fix histogram processing

### PR DESCRIPTION
**Pull request by cabreraalex**
# Description
Simplify histogram generation, only one function to generate buckets and another to calculate counts & metrics.

### CODE REVIEW

File: frontend/src/lib/components/metadata/HistogramsHeader.svelte [hunk 1]
```diff
@@ -8,7 +8,7 @@
 </script>
 
 <div
-	class="flex items-center justify-between pt-2.5 sticky top-24 border-b border-grey-lighter bg-yellowish-light z-10 pb-4"
+	class="flex items-center justify-between pt-2.5 top-24 border-b border-grey-lighter bg-yellowish-light z-10 pb-4"
```
Comment (Sparkier) [inline frontend/src/lib/components/metadata/HistogramsHeader.svelte:11]: Not sure but if not sticky can we remove the top property as well?

File: frontend/src/lib/components/general/Header.svelte [hunk 2]
```diff
@@ -70,7 +70,7 @@
 			{/if}
 		</div>
 		<div class="flex flex-col items-center justify-center mb-3">
-			{#if currentTab?.includes('project') && $project && $project.ownerName === user?.name}
+			{#if (currentTab?.includes('explore') || currentTab?.includes('compare')) && $project?.ownerName === user?.name}
```
Comment (Sparkier) [inline frontend/src/lib/components/general/Header.svelte:73]: Don't we want to show this for chart? Seems like something people might want to access after chart creation.

**Review by Sparkier — APPROVED** [review]

File: backend/zeno_backend/database/util.py [hunk 3]
```diff
@@ -18,6 +18,8 @@ def resolve_metadata_type(data_frame: pd.DataFrame, column: str) -> MetadataType
     """
     dtype = data_frame[column].dtype
     if pd.api.types.is_any_real_numeric_dtype(dtype):
+        if data_frame[column].nunique() < 20:
```
[the remark this answers is not in the document]
Comment (cabreraalex) [reply backend/zeno_backend/database/util.py:21]: ugh I know it's terrible let me think a bit more about it. There are so many weird edge cases...

File: frontend/src/lib/components/general/Header.svelte (hunk 2 above)
Comment (cabreraalex) [reply frontend/src/lib/components/general/Header.svelte:73]: Good catch, it wasn't working at all before when checking for project.

File: backend/zeno_backend/database/util.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (cabreraalex) [reply backend/zeno_backend/database/util.py:21]: idk how to do this well, going to make a linear issue.

**Pull request merged**